numba/core/lowering.py

```diff
@@ -1281,10 +1284,11 @@ def _alloca_var(self, name, fetype):
             # quit early
             return
 
-        llty = self.context.get_value_type(fetype)
-        # If the name is used in multiple blocks...
-        if name not in self._singly_assigned_vars:
+        # If the name is used in multiple blocks or lowering with debuginfo...
+        flags = utils.ConfigStack.top_or_none()
+        if (name not in self._singly_assigned_vars) or flags.debuginfo:
```

Comment:
There's about 6 checks for debug info added in this file - whilst its clear at the time of this review that these are all added to disable the SROA-like optimization in debug mode, I think later down the line looking at these functions in isolation it might not be all that clear why these conditions include `flags.debuginfo` or `not flags.debuginfo` - is there a way to replace the use of this flag at each site with a variable that makes it clear what its reason is - e.g. named `disable_sroa_like_opt` (or something more apt, I'm presently lacking in imagination)?

Reply:
Good point, fixed in 8329107